user-console/src/main/resources/org/pentaho/mantle/public/browser/js/browser.js

```diff
@@ -1555,10 +1576,72 @@ define([
 
         if (elementBottom > scrollBottom) {
           listboxNode.scrollTop = elementBottom - listboxNode.clientHeight;
-        } else if (selectedOption.offsetTop < listboxNode.scrollTop) {
-          listboxNode.scrollTop = selectedOption.offsetTop;
+        } else if ((selectedOption.offsetTop - listboxNode.offsetTop) < listboxNode.scrollTop) {
+          listboxNode.scrollTop = selectedOption.offsetTop - listboxNode.offsetTop;
+        }
+
+      }
+    },
+
+    findItemToFocus: function (key, activeDescendant, listboxNode) {
+      var itemList = listboxNode.querySelectorAll('[role="option"]');
+      var character = String.fromCharCode(key);
+      var searchIndex = 0;
+
+      if (!this.model.get("keysSoFar")) {
+        for (var i = 0; i < itemList.length; i++) {
+          if (itemList[i].getAttribute('id') === activeDescendant) {
+            searchIndex = i;
+          }
         }
+      }
+      var keysSoFar = this.model.get("keysSoFar") + character;
+      this.model.set("keysSoFar", keysSoFar);
+      this.clearKeysSoFarAfterDelay(this.model.get("keysSoFar"));
 
+      var nextMatch = this.findMatchInRange(itemList, searchIndex + 1, itemList.length);
+      if (!nextMatch) {
+        nextMatch = this.findMatchInRange(itemList, 0, searchIndex);
+      }
+      return nextMatch;
+    },
+
+    findMatchInRange: function (list, startIndex, endIndex) {
+      // Find the first item starting with the keysSoFar substring, searching in
+      // the specified range of items
+      for (var n = startIndex; n < endIndex; n++) {
+        var label = list[n].innerText;
+        if (label && label.toUpperCase().indexOf(this.model.get("keysSoFar")) === 0) {
+          return list[n];
+        }
+      }
+      return null;
+    },
+
+    clearKeysSoFarAfterDelay: function () {
+      if (keyClear) {
+        clearTimeout(keyClear);
+        keyClear = null;
+      }
+      var keyClear = setTimeout(
+        function () {
+          this.model.set("keysSoFar", '');
+          keyClear = null;
+        }.bind(this),
+        500
+      );
+    },
+
+    onBlur: function (event) {
+      var activeDescendant = event.currentTarget.getAttribute('aria-activedescendant');
+      if (activeDescendant) {
+        var currentItem = document.getElementById(activeDescendant);
+        currentItem.classList.remove('navigatebykey');
+        if ($(event.currentTarget).find('.selected').length > 0) {
+          event.currentTarget.setAttribute('aria-activedescendant', $(event.currentTarget).find('.selected')[0].id);
+        } else {
+          event.currentTarget.setAttribute('aria-activedescendant', '');
+        }
```

Comment:
Please, see discussion in https://github.com/pentaho/pentaho-platform-plugin-common-ui/pull/1642#discussion_r1162982156.

It would be best to use the same CSS class (instead of `navigatebykey`, the proposed `active-descendant`). I believe the associated CSS rules are compatible. See https://github.com/pentaho/pentaho-commons-gwt-modules/pull/889/files#diff-8f4efda5dfb33a49cb02642aa3b6b9b3d5cb98a65d30b691ad4e0e3cbc9b2f3eR5164.

Also, as there explained, the focus/blur events should not be needed to keep this class up-to-date — just update at the same places and time where/when `aria-activedescendant` is updated.